erts/emulator/beam/erl_math.c

```diff
@@ -293,3 +294,82 @@ BIF_RETTYPE math_fmod_2(BIF_ALIST_2)
 {
     return math_call_2(BIF_P, fmod, BIF_ARG_1, BIF_ARG_2);
 }
+
+BIF_RETTYPE math_nearbyint_1(BIF_ALIST_1)
+{
+    Eterm res;
+    FloatDef f;
+
+    /* check arg */
+    if (is_not_float(BIF_ARG_1)) {
+	if (is_integer(BIF_ARG_1)) {
+	    BIF_RET(BIF_ARG_1);
+	}
+	BIF_ERROR(BIF_P, BADARG);
+    }
+
+    /* get the float */
+    GET_DOUBLE(BIF_ARG_1, f);
+
+    /* Round it and return the resultant integer. Note that the current rounding mode
+     * must be "to nearest" for `nearbyint` to tie to even. */
+    ASSERT(1 == FLT_ROUNDS);
```

Comment:
We've decided to take this in, but we'd like the rounding mode to be set to `FE_TONEAREST` and then restored before returning. The initial value is implementation-defined and we can't rely on setting it on boot as NIFs and drivers can change it at will.

https://en.cppreference.com/w/c/numeric/fenv/feround

Reply:
Thank you! Quick question: is changing it thread-safe?

Reply:
Yes, it's thread-local.

Reply:
In any case, I have pushed the requested update!